### DISCUSSION TITLE
CSV overwrite for additional fields

### DISCUSSION THREAD

**Discussion by Pletoo** [sheltermanager/asm3#1528] [Q&A / Help] [answered]
Do our additional fields get  overwritten when importing a CSV with an existing record? Would it be very difficult to make it an option when creating an additional field to indicate whether it can be overwritten or not?

One of our additional fields is a description of the collar they are wearing, but when a kitten grows out of it's collar, it gets changed. I'd like to update all  at once rather than individually. On the other hand, I have a field to indicate the date when the initial vaccines are complete that I definitely do not want to overwrite accidentally.

**Comment by bobintetley** [comment, marked as answer]
Yes, additional fields that are specified again in a file will replace the current value. If you have the column in your file for the additional field, but leave the value blank, it won't remove the existing value during import.

It sounds like you're trying to replace the UI with CSV imports?

**Reply by Pletoo** [reply]
Not at all - just trying to spend less time manually inputting information on our additional fields without messing something up.

Yesterday I had to process 32 kittens that went to the vet. I used the bulk change to put in the altering and bulk vaccine to put in the rabies shot as due and given - worked wonderfully. However the 32 different collars had to go in by CSV import. In about 3 weeks we will have 30 more to take in, plus the ones yesterday will need different collars if they haven't been adopted. The collars will always have to be CSV so I wanted to be certain that it would overwrite when I change collars. I didn't find that answer in the manual.

The "Initial Vaccines Complete" field marks when the entire series of vaccines has been given (Rabies plus 1-3 Distemper shots.) The vaccines have been incorrectly recorded for over 4 years by my predecessors and co-workers. Even without correcting records of cats that are no longer at the center, it still is very time consuming to fix. I run a report showing history for a group of cats and when I have manually gone through their paper record and verified that the information has been corrected, I fill in that additional field with the date of their final shot, imported in bulk via CSV. This gives me a way to filter them out of the next report. I definitely do not want to overwrite that and have to start over. Since this is not something that anyone else needs, I am using the UI functionality that allows me to create an additional field and import a CSV.

The vaccine book simply does not fully work for us at the moment until I can get everything cleaned up. It is a great tool to look at shots given recently and as a reminder when 1 year boosters are due for the handful of cats that are still with us or have been returned. However, since initial rabies is always given at the vet and I have a printed list that I refer to when completing the bulk change for altering, I just use that same list to create and give a bulk vaccine. 

Yes, I am aware that if the rabies were pre-scheduled and in the book, I could just select them and use the give function and schedule the 1 year booster in one move. I will eventually re-establish the templates to do that when a new animal is added, but I paused that and deleted all of the upcoming shots because I had to get ahead of it all. When I first started, there were over 580 vaccines "due" because every single one had been input incorrectly with one date instead of 2. It actually got more complicated after I "fixed" the issue by creating templates that put the first date in. Then some were correct and some were incorrect because the wrong entry category was used or because a duplicate entry was created.

Distemper just has too many variables for it to be worthwhile to schedule the initial shot with a template or otherwise. I don't see that changing so that will continue to be something I track on a spreadsheet and use a CSV to update. Eventually I will be able to confidently use the book for Distemper boosters. 

For the record, I think SM is a fantastic program. I can't even begin to imagine how I would keep track of most of the information without it, and there are so many details that I would never have thought of on my own. I have pretty much read through everything in the manual and this forum, and have done extensive testing using the demo and a test user account. Most of the time if I am asking a question it is because I haven't found an answer (sometimes it is a tired moment when I ask something and realize later I knew the answer or could have tested it myself.)  I give feedback and make suggestions because I like the program - if I didn't, I wouldn't bother.
